lib/adal-aad-node/src/engine/AzureAuth.ts

```diff
@@ -0,0 +1,458 @@
+		if (shouldOpen) {
+			const result = await this.login(tenant, resource);
+			result?.authComplete?.resolve();
+			return result?.response;
+		}
+		return undefined;
+	}
+
+	/**
+	 * Asks the user if they would like to do the interaction based authentication as required by OAuth2
+	 * @param tenant
+	 * @param resource
+	 */
+	private async askUserForInteraction(tenant: Tenant, resource: AADResource): Promise<boolean> {
+		return this.userInteraction.askForConsent(this.stringLookup.getInteractionRequiredString({ tenant, resource }));
+
+	}
+	//#endregion
+
+	//#region data modeling
+
+	public createAccount(tokenClaims: TokenClaims, key: string, tenants: Tenant[]): AzureAccount {
+		// Determine if this is a microsoft account
+		let accountType: 'microsoft' | 'work_school';
+
+		if (tokenClaims?.idp === 'live.com') {
+			accountType = 'microsoft';
+		} else {
+			accountType = 'work_school';
+		}
+
+		const name = tokenClaims.name ?? tokenClaims.email ?? tokenClaims.unique_name;
+		const email = tokenClaims.email ?? tokenClaims.unique_name;
+
+		let displayName = name;
+		if (email) {
+			displayName = `${displayName} - ${email}`;
+		}
+
+		const account: AzureAccount = {
+			key: {
+				providerId: this.providerSettings.id,
+				id: key,
+				accountVersion: AzureAuth.ACCOUNT_VERSION,
+			},
+			displayInfo: {
+				accountType,
+				userId: key,
+				displayName,
+				email,
+				name,
+			},
+			properties: {
+				providerSettings: this.providerSettings,
+				isMsAccount: accountType === 'microsoft',
+				tenants,
+				azureAuthType: this.azureAuthType
+			},
+			isStale: false
+		};
+
+		return account;
+	}
+
+	//#endregion
+
+	//#region network functions
+	public async makePostRequest(url: string, postData: AuthorizationCodePostData | TokenPostData | DeviceCodeStartPostData | DeviceCodeCheckPostData): Promise<AxiosResponse<any>> {
+		const config: AxiosRequestConfig = {
+			headers: {
+				'Content-Type': 'application/x-www-form-urlencoded'
+			},
+			validateStatus: () => true // Never throw
+		};
+
+		// Intercept response and print out the response for future debugging
+		const response = await axios.post(url, qs.stringify(postData), config);
+		this.logger.pii(url, postData, response.data);
+		return response;
+	}
+
+	private async makeGetRequest(url: string, token: string): Promise<AxiosResponse<any>> {
+		const config: AxiosRequestConfig = {
+			headers: {
+				'Content-Type': 'application/json',
+				'Authorization': `Bearer ${token}`
+			},
+			validateStatus: () => true // Never throw
+		};
+
+		const response = await axios.get(url, config);
+		this.logger.pii(url, response.data);
+		return response;
+	}
+
+	//#endregion
+
+	//#region utils
+	protected getTokenClaims(accessToken: string): TokenClaims | undefined {
+		try {
+			const split = accessToken.split('.');
+			return JSON.parse(Buffer.from(split[1], 'base64').toString('binary'));
+		} catch (ex) {
+			throw new Error('Unable to read token claims: ' + JSON.stringify(ex));
```

Comment:
For all the errors - are you ensuring these are being logged somewhere too?

Reply:
The caller needs to catch them basically

Reply:
The library should log all errors though I would think. So yeah - leave it up to the caller to handle (and log themselves if they want) but if an error happens we should make sure that the adal log file has an entry for that too.